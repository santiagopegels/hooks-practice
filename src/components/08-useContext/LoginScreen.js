import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {

    const { setUser } = useContext(UserContext)

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr></hr>

            <button
                className="btn btn-primary"
                onClick={() => setUser({
                    id: 1123,
                    name: 'Test'
                })}
            >
                Login
            </button>
        </div>
    )
}
