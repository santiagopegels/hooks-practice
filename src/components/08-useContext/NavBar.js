import React from 'react'
import { NavLink } from 'react-router-dom'

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-lg navbar-light bg-light">
        <div className="container-fluid">
          <a className="navbar-brand" href="#">useContext</a>
          <div className="collapse navbar-collapse" id="navbarNav">
            <div className="navbar-nav">
              <li className="nav-item">
                <NavLink exact to='/' activeClassName="active" className="nav-link">Home</NavLink>
              </li>
              <li className="nav-item">
              <NavLink exact to='/login' activeClassName="active" className="nav-link">Login</NavLink>
              </li>
              <li className="nav-item">
              <NavLink exact to='/about' activeClassName="active" className="nav-link">About</NavLink>
              </li>
            </div>
          </div>
        </div>
      </nav>
    )
}
