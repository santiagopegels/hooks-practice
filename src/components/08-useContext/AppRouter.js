import React from 'react'
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Link,
    Redirect
} from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <Router>
            <div>
               <NavBar />

                {/* A <Switch> looks through its children <Route>s and
              renders the first one that matches the current URL. */}
                <Switch>
                    <Route exact path="/about" component={AboutScreen} />
                    <Route exact path="/login" component={LoginScreen} />
                    <Route exact path="/" component={HomeScreen} />
                    
                    <Redirect to="/" />
                </Switch>
            </div>
        </Router>
    )
}
