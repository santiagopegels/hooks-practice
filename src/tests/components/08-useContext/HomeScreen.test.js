import React from 'react'
import { mount } from 'enzyme'
import { HomeScreen } from '../../../components/08-useContext/HomeScreen'
import { UserContext } from '../../../components/08-useContext/UserContext'

describe('Tests on <HomeScreen />', () => {

    const user = {
        name: 'Test',
        email: 'test@test.com'
    }

    const wrapper = mount(
        <UserContext.Provider value={{ user }}>
            <HomeScreen />
        </UserContext.Provider>
    )

    test('should show correctly', () => {

        expect(wrapper).toMatchSnapshot()

    })

})
