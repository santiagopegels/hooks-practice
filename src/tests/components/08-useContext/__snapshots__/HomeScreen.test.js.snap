// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <HomeScreen /> should show correctly 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Test",
   "email": "test@test.com"
}
  </pre>
</div>
`;
